### ISSUE TITLE
supabase oauth azure provide error : Unable to Exchange External code 

### ISSUE THREAD

**Issue by subhaschnadraBodaki** [supabase/supabase#4640]
# Bug report

## Describe the bug
I used azure external oauth provider to log in

Redirect url used is ------ https://aomnqngcqjdcvqovmjjn.supabase.co/auth/v1/callback

It is redirecting me to -------- http://localhost:3000/?error=server_error&error_description=Unable+to+exchange+external+code%3A+0.AVYATe-lM3aaxE2G3zQy-ilZg3WsJb7sWJpBnTQLR_-RJx1WAEU.AQABAAIAAAD--DLA3VO7QrddgJg7Wevrpsz8QdQCyeSynIokYb7cJhztFsxBEpVABWWdpY29U2QtxQtwbiXzdtFS-UnW5EFzSAKiJZIp8gTarIR2YW_jE70aT71TvQ_4-v3JXZi_S_pKYBKF_12fPYks1OuJVDItxEc6MHyOd8iI7HZtwzO-44hAMDBOGvvyw9rszE7Mg-Y6heN9_DuPS8lTgTavwYG5ekyW2W4121LXl8HH0qTF9kCusbVQfLPo8OUOnXX-LwgOfvIuuZm1UmGGSEjddSDa5PCRplKbu9168odz9oywD33p_dw2bQLmWDWHqJd7Q6NWJ1zsEQB396Y_dwABGnKfnNVOPQ3r0XhaDWMkGS2iFLAzUUBSwJtMR7yoGs8MjutNKZhjosPQp_79c86dIvKbzg3E_yKfiEFyZGwK7WALHnVDnmzz1m4FaMHMYPi-rQTNhVoWFFkqsdiQxciUmHOUK6Ul_RgYj-Vx7F4h1dq7N3BZFJEcKaxo9rBMv8I6oSPzCkd9vTbNVXRr96F3uI9ZJIS23ajsy_a2oTb_v3xDgu77V5-GJ6-NuEIvaq0q59Qj7bONwM2d11cZmZrnR5jpNi9ha99DqSMhWgU3HDZeiPwKqrbd3Mrk5x-jxeBbZPopBdYAit21sp7HSHPlp2exIAA#

When i tried it with google it is working fine
I checked the space also there is no space while entering client id and secret

Can You pls look at this
after Logging i am reflected to login page again
![Screenshot (3)](https://user-images.githubusercontent.com/85378633/147224586-5d2b288c-37e8-4581-b799-78ea4d42d542.png)

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to '…'
2. Click on '…'
3. Scroll down to '…'

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of supabase-js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

**Comment by kiwicopple** [comment]
Hey @subhaschnadraBodaki , did you follow all the instructions here to set up your Azure app?

https://github.com/supabase/gotrue/pull/54#issuecomment-757043573

**Comment by mike-hogan** [comment]
Hey @subhaschnadraBodaki, I had that error too until (in Azure portal) I configured the redirect uri as a Web app redirect uri, instead of an SPA (single page app) redirect uri.  Then that error disappeared, but the next error I got was "Error getting user email from external provider"

**Comment by kiwicopple** [comment]
> Error getting user email from external provider

Does your Azure app allow permissions to get the users' email address?

**Comment by mike-hogan** [comment]
> Does your Azure app allow permissions to get the users' email address?

Yes, my app has the same 5 scopes as shown in https://github.com/supabase/gotrue/pull/54
